### DISCUSSION TITLE
How to link BaseVar and its real value

### DISCUSSION THREAD

**Discussion by TaiJuWu** [reflex-dev/reflex#1147] [Q&A] [answered]
Hello every,

I have a question about var and its value.
When I write code below
```python
class State(pc.State):
    """The app state."""
    test = [1,2,3]
```

The variable become BaseVar(name='test'.....) in app.init_subclasee
```python
        cls.base_vars = {
            f.name: BaseVar(name=f.name, type_=f.outer_type_).set_state(cls)
            for f in cls.get_fields().values()
            if f.name not in cls.get_skip_vars()
        }
        cls.computed_vars = {
            v.name: v.set_state(cls)
            for v in cls.__dict__.values()
            if isinstance(v, ComputedVar)
        }
        cls.vars = {
            **cls.inherited_vars,
            **cls.base_vars,
            **cls.computed_vars,
        }
        cls.event_handlers = {}
        print(cls.vars) # My add
```
But when we use the compiler, it need real value([1,2,3]).
I can't understand how to link BaseVar(name="test"...) and real value([1,2,3])  in pynecone?

If anyone can help me, I will be grateful.

**Comment by masenf** [comment, marked as answer]
@TaiJuWu the State attribute is a `BaseVar` when it is accessed from the class, i.e. `State.test`. This is used when creating the frontend: the actual value is in the state dict on the client side and will be filled in by the BaseVar at runtime.

The value contains the actual list when it is accessed from the instance, i.e. from an event handler. This is called when some client interaction triggers an event on the state. Within the event handler, accessing `self.test` will get the real `list` value, not the `BaseVar`.

In the context where you have a reference to the `State` class, during compile time, it doesn't make any sense to access the actual value, because the page is compiled for _all clients_, but the actual state value is _per-client_, so it only makes sense to access in the context of an event handler.

**Reply by TaiJuWu** [reply]
Oh, I see.
I haven't understood the event handler part yet
I will pay special attention to this part.
Thank you very much!!!
